linker/linker_soinfo.h

```diff
@@ -338,6 +340,40 @@ struct soinfo {
   uintptr_t handle_;
 
   friend soinfo* get_libdl_info(const char* linker_path, const link_map& linker_map);
+
+ public:
+  struct SegmentInfo {
+    ElfW(Addr) phdr_addr, real_addr;
+    ElfW(Word) real_size, page_size;
+    size_t index;
+  };
+
+  typedef std::vector<SegmentInfo> seginfo_list_t;
```

Comment:
Having a typedef makes the uses shorter and more readable (you just put in `seginfo_list_t` or whatever instead of `std::vector<SegmentInfo>`).